pkg/util/k8s/pod.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import v1 "k8s.io/api/core/v1"
+
+func IsPodTerminal(pod *v1.Pod) bool {
```

Comment:
IsPodTerminated?

Reply:
I saw this function name in kubernetes several times, e.g. https://github.com/kubernetes/kubernetes/blob/release-1.30/staging/src/k8s.io/endpointslice/util/controller_utils.go#L95-L98.

I just found there is a public util function `IsPodTerminal` in "k8s.io/kubernetes/pkg/api/v1/pod" and changed to use it.

Reply:
Just realized this is in an internal package of kubernetes, so still have to define our own one.